cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -2052,13 +2052,17 @@ fn x64_get_operands<F: Fn(VReg) -> VReg>(inst: &Inst, collector: &mut OperandCol
             mem.get_operands(collector);
         }
 
-        Inst::AtomicRmwSeq { .. } => {
-            // FIXME: take vreg args, not fixed regs, and just use
-            // reg_fixed_use here.
-            collector.reg_use(regs::r9());
-            collector.reg_use(regs::r10());
-            collector.reg_def(Writable::from_reg(regs::r11()));
-            collector.reg_def(Writable::from_reg(regs::rax()));
+        Inst::AtomicRmwSeq {
+            address,
+            operand,
+            temp,
+            dst_old,
+            ..
+        } => {
+            collector.reg_fixed_use(*address, regs::r9());
+            collector.reg_fixed_use(*operand, regs::r10());
```

Comment:
These two should be "late uses", or otherwise the below two defs should be "early defs", semantically. It won't matter yet with the fixed reg constraints (neither of the defs could possibly overlap either of the uses) but it will matter when we eventually remove the need for the fixed regs, and I'd rather note it now while we're thinking about it!

I don't think there is a `reg_fixed_late_use` on the `OperandCollector` but we should be able to add one, analogously to the other helpers there.

Reply:
I think we probably want a `fixed_reg_use_at_end` helper in `regalloc2`. See what I did in `OperandCollector` in the interim.